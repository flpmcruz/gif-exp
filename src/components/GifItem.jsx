import PropTypes from 'prop-types'

export const GifItem = ({title, url, id}) => {
  return (
    <div className="card">
        <img src={ url } alt={ title }></img>
        <p>{ (title.length <= 1) ? "Sin titulo" : title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string,
}