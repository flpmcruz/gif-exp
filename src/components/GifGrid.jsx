import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando</h2>}
      
      {/* otra forma de hacerlo */}
      {/* {isLoading ? <h2>Cargando</h2> : null} */}

      <div className="card-grid">
        {images.map((img) => (
          <GifItem
            key={img.id}
            //de esta forma envio todas las propiedades esparcidas
            {...img}
          />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}