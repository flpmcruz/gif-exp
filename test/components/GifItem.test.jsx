import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba de GifItem", () => {

  const title = "Hola soy un titulo";
  const url = "https://media4.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=790b7611e464637aa4f68dcd0ae2195408b5514d9a482640&rid=giphy.gif&ct=g";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    
    expect( container ).toMatchSnapshot();
  });

  test("Debe mostrar la imagen de la URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug()

    //expect( screen.getByRole('img').src).toBe(url)
    //expect( screen.getByRole('img').alt).toBe(title)
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( alt )
  });

  test('Debe mostrar el titulo en el componente', () => { 
    render(<GifItem title={title} url={url} />);
    expect( screen.getByText( title ) ).toBeTruthy()
   })

});
