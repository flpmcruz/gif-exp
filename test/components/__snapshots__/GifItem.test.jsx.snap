// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola soy un titulo"
      src="https://media4.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=790b7611e464637aa4f68dcd0ae2195408b5514d9a482640&rid=giphy.gif&ct=g"
    />
    <p>
      Hola soy un titulo
    </p>
  </div>
</div>
`;
